docs/examples/types_generics.py

```diff
@@ -0,0 +1,81 @@
+from pydantic import BaseModel, ValidationError
+from pydantic.fields import ModelField
+from typing import TypeVar, Generic
+
+OldType = TypeVar('OldType')
+GradeType = TypeVar('GradeType')
+
+# This is not a pydantic model, it's an arbitrary generic class
+class Assessment(Generic[OldType, GradeType]):
```

Comment:
I don't think I understand what an "assessment" is in this scenario, why do it have a thing called "old"?

I know it doesn't really matter but it makes the example easier to read if you can imagine what this is.

Reply:
Yeah, totally agree. I was trying to find a scenario where the same model could have different types for the same field name, and have at least 2 of these fields with a type from vars.

My idea (clearly not great) was that an "assessment" could be used to measure the age and school grade of a student, and at the same time could be used to evaluate the features and quality of cheese, if it's "old" being a boolean and a string for the grade.

Let me try and make it for wine and cheese, not students. And add some extra verbal explanation of the intention.

Reply:
Ended up very good, demonstrated how powerful (and perhaps mind bending) generics can be.